pkg/scheduler/scheduler.go

```diff
@@ -639,6 +639,10 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 			fwk.RunReservePluginsUnreserve(bindingCycleCtx, state, assumedPod, scheduleResult.SuggestedHost)
 			if forgetErr := sched.SchedulerCache.ForgetPod(assumedPod); forgetErr != nil {
 				klog.ErrorS(forgetErr, "scheduler cache ForgetPod failed")
+			} else {
+				// "Forget"ing an assumed Pod in binding cycle should be treated as a PodDelete event,
+				// as the assumed Pod had occupied a certain amount of resources in scheduler cache.
+				sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, nil)
```

Comment:
> What about the assumed pod itself? If it's unschedulable, it shouldn't be retried, but if it's another error, it should.

Good point. To make the logic 100% precise, we should exclude the pod itself by adding a precheckFunc - something like:

```go
sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalquAssignedPodDelete, func(pod *v1.Pod) bool {
	return assumedPod.UID != pod.UID
})
```

> Also, can we do this as a defer instead of calling it everywhere?

That's also something I wanted to do, not only this logic, but also other duplicated pieces. We can do a it in a follow-up.

Reply:
> Applies for any other errors below.

@alculquicondor I checked the code. It seems [RunPreBindPlugins()](https://github.com/kubernetes/kubernetes/blob/603830b8a694546aa74ea33175864fb6f6858067/pkg/scheduler/framework/runtime/framework.go#L861-L865) and [RunBindPlugins()](https://github.com/kubernetes/kubernetes/blob/603830b8a694546aa74ea33175864fb6f6858067/pkg/scheduler/framework/runtime/framework.go#L894-L898) will unconditionally wrap the plugin status as an error status, and this behavior has existed for a while. So I'm thinking we'd better only care about excluding the assumed Pod for **permit case** only. WDYT? (otherwise we need to tweak the error aggregating logic in prebind and bind)

Reply:
Well, for WaitPermit, it's always Unschedulable...

https://github.com/kubernetes/kubernetes/blob/f8ef5a9411595173f2ac3a510713dcabef9be297/pkg/scheduler/framework/runtime/waiting_pods_map.go#L162

Reply:
Then, conclusion is: for Permit failures, exclude the incoming Pod from the move. For PreBind/Bind failures, include the incoming Pod on the move.

I think we don't have to do this in `MoveAllToActiveOrBackoffQueue` because the Pod is not currently in the queues. We would add it after we call this method in `recordSchedulingFailure`. Here, we have to fix what we discussed offline: if internal  error, add the Pod to Backoff instead of Unschedulable.

Reply:
> I think we don't have to do this in `MoveAllToActiveOrBackoffQueue` because the Pod is not currently in the queues.

That's true. However, a bit tricky logic here is: once we call MoveAllToActiveOrBackoffQueue(), the queue's `moveRequestCycle` will be updated:

https://github.com/kubernetes/kubernetes/blob/8088b3e67d3f917a94b4ac530579c22cd7688fe6/pkg/scheduler/internal/queue/scheduling_queue.go#L631

And then upon `recordSchedulingFailure`, the assumed pod will be moved to backoffQ anyways:

https://github.com/kubernetes/kubernetes/blob/8088b3e67d3f917a94b4ac530579c22cd7688fe6/pkg/scheduler/internal/queue/scheduling_queue.go#L398-L400

I will append a commit to give a workaround.